src/lib/prisma-session-store.ts

```diff
@@ -0,0 +1,440 @@
+
+      try {
+        await this.prisma.session.deleteMany();
+
+        if (callback) defer(callback);
+      } catch (e: unknown) {
+        if (callback) defer(callback, e);
+      }
+    };
+
+    /**
+     * Get the count of all sessions in the store
+     *
+     * @param callback
+     */
+    public readonly length = async (
+      callback?: (err?: unknown, length?: number) => void
+    ) => {
+      if (!(await this.validateConnection())) return callback?.();
+
+      // XXX More efficient way? XXX
+
+      try {
+        const sessions = await this.prisma.session.findMany({
+          select: { sid: true }, //Limit what gets sent back; can't be empty.
+        });
+
+        const itemCount = sessions.length;
+        if (callback) defer(callback, undefined, itemCount);
+        return itemCount;
+      } catch (e: unknown) {
+        if (callback) defer(callback, e);
+      }
+    };
+
+    /**
+     * Destroy the session associated with the given `sid`(s).
+     *
+     * @param sid a single or multiple id(s) to remove data for
+     * @param callback a callback notifying that the session(s) have
+     * been destroyed or that an error occurred
+     */
+    public readonly destroy = async (
+      sid: string | string[],
+      callback?: (err?: unknown) => void
+    ) => {
+      if (!(await this.validateConnection())) return callback?.();
+
+      if (Array.isArray(sid)) {
+        await Promise.all(sid.map(async (s) => this.destroy(s, callback)));
+      } else {
+        try {
+          await this.prisma.session.delete({ where: { sid } });
+        } catch (e: unknown) {
+          this.logger.warn(
+            `Attempt to destroy non-existent session:${sid} ${e}`
+          );
+          if (callback) defer(callback, e);
+        }
+      }
+
+      if (callback) defer(callback);
+    };
+
+    /**
+     * Remove only expired entries from the store
+     */
+    public readonly prune = async () => {
+      if (!(await this.validateConnection())) return;
+
+      // XXX More efficient way? Maybe when filtering is fully implemented? XXX
+
+      this.logger.log('Checking for any expired sessions...');
+      const sessions = await this.prisma.session.findMany({
+        select: {
+          expires: true,
+          sid: true,
+        },
+      });
+
+      for (const session of sessions) {
+        const now = new Date();
+        const remainingSec = (session.expires.valueOf() - now.valueOf()) / 1000;
+        this.logger.log(`session:${session.sid} expires in ${remainingSec}sec`);
+
+        if (now.valueOf() >= session.expires.valueOf()) {
+          this.logger.log(`Deleting session with sid: ${session.sid}`);
+          await this.prisma.session.delete({
+            where: { sid: session.sid },
+          });
+        }
+      }
+    };
+
+    /**
+     * Start an interval to prune expired sessions
+     */
+    public startInterval(): void {
+      const ms = this.options.checkPeriod;
```

Comment:
🤔  I was thinking that the `checkPeriod` should be off by default. So that would be `0` if not defined in the options.

Reply:
@wSedlacek - Have to step away for the day here, but hope to get back to this in the evening or tomorrow morning.

Reply:
The code is definitely simpler if `checkPeriod` is "off" by default.

I do wonder about the use-cases though... If someone _forgets_ to set a value for checkPeriod: Is it more problematic for sessions persist indefinitely, or for sessions to be removed at some "canned" default interval?

**If we assume most end users are on mobile devices: Well, mobile "sessions" typically last indefinitely**, because a) Early on, Android and iOS incorporated login screens as a default (where perhaps Windows and OSX didn't? Not 100% sure of this) - resulting in an extra security layer, and b) Typing in UN/PWD is more annoying to do on a touchscreen.

**If we assume most end-users are on laptop/desktop devices: Again, not 100% sure, but I suspect that session times are typically a couple days to a month, conventionally.** If a developer simply forgets to set a value for check period, do we want the conventional behavior to happen, or do we want to demand that the developer be explicit?

How about you make the call. Mainly, I just wanted to explore what things might look like from mobile/desktop end-user perspectives/expectations here, in case it introduces anything new for your consideration.

Reply:
Remember that `checkPeriod` checks the DB to remove sessions.
However the cookie for the session still expires, so the only thing having this turned off would do is cause the DB to have a bunch of old sessions.